ProfileManager.js

```diff
@@ -115,6 +128,46 @@ export default class ProfileManager {
     return await this._getAgentContent({profileAgentRecord});
   }
 
+  /**
+   * Gets a capability specified by its reference ID for profile agent from
+   * a cache or delegates one on the fly if it does not exist.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} options.id - The reference ID of the zcap.
+   * @param {string} options.profileAgent - The profile agent requesting a zcap.
+   * @param {boolean} options.useEphemeralSigner - Flag to enable invoking
+   *   capabilities with the ephemeral invocation signer associated with the
+   *   currently authenticated session, default `true`. See more in class
+   *   description.
+   *
+   * @returns {Promise<object>} The capability for the profile agent.
+   */
+  async getAgentCapability({id, profileAgent, useEphemeralSigner = true}) {
+    const capabilityCacheKey = `profileAgent-${profileAgent.id}-zcaps`;
+    const capabilityKey = `${capabilityCacheKey}-${id}-${useEphemeralSigner}`;
+
+    if(!this._cacheContainer.has(capabilityCacheKey)) {
+      this._cacheContainer.set(capabilityCacheKey, new Cache());
+    }
+
+    const capabilityCache = this._cacheContainer.get(capabilityCacheKey);
+    const agentZcap = capabilityCache.get(capabilityKey);
+    if(agentZcap) {
+      return agentZcap;
+    }
+
+    const promise = this._getAgentCapability(
+      {id, profileAgent, useEphemeralSigner});
```

Comment:
Does this fit in 80 chars?

```suggestion
    const promise = this._getAgentCapability({id, profileAgent, useEphemeralSigner});
```

Reply:
No.